### ISSUE TITLE
Organizing OTP's

### ISSUE THREAD

**Issue by Bartvdgriendt** [tijme/raivo#9]
The search function is great but I think it would be a nice addition to be able to organize the OTPs. With this option I would prioritize them from most frequently used to least frequently used. For the OTP's that I use less frequently I would use the search function.

An option would be something like this:

![ALT](https://camo.githubusercontent.com/6bb369cfd75351527be85e91482c468c51ae1f16/687474703a2f2f6d656469612e69646f776e6c6f6164626c6f672e636f6d2f77702d636f6e74656e742f75706c6f6164732f323031342f30332f694f532d372d4d61696c2d7265617272616e67696e672d73686f7274637574732e676966)

**Comment by tijme** [comment]
This is a nice feature request. One of my colleagues asked for it too. 

I'm currently thinking of building some sort of "Pin to top" feature that allows you to pin some of your OTP's to the top (and also allows you to order your pinned OTP's manually). Your unpinned OTPs will then still be sorted alphabetically.

Would that fit your needs?

**Comment by Bartvdgriendt** [comment]
Yes, that would definitely suite my needs! I only use a couple of OTP's on a daily basis so a feature like that would be great!

**Comment by nolageek** [comment]
Please add categories/folders. I'm currently using multiple 2FA apps for. Two for work and one for personal. The only reason I'm doing this is because hardly any 2FA apps allow me to keep things separate. Across these apps I have over 60 accounts added and it's a pain when they're all mixed together in a single one.

**Comment by EspadaV8** [comment]
AndOTP Has a nice feature where it sorts but most used entries. It would be great to have something like that.

**Comment by a-was** [comment]
I'm also waiting for this. I guess this can be helpful
https://programmingwithswift.com/move-list-item-with-swiftui/

**Comment by steffenrapp** [comment]
I also like the idea with folders or categories to group different accounts as personal or work etc. Would defintely make it much easier to find the right OTP.

**Comment by meichthys** [comment]
Related to #92

**Comment by jfreedman0** [comment]
Would love to be able to just sort by alphabetical too rather than having them only in the order they were created.

**Comment by nokje** [comment]
I'd benefit from categorising my OTP's aswel. Imagining a way of organising OTP's by organisational units (OU) like "Private", "Work" or "Hobby project B" wherein prioritising could be done by pinning or last/most used OTP. Additionally for UX, swiping UP or DOWN with 2 fingers to shift to/switch to the OU containing the OTP's where each OU has a distinctive theme eg. a different background color.

**Comment by Archer36** [comment]
I would love the ability to at least pin my favorites to the top and/or manually sort. @tijme you mentioned you were considering implementing a "pin to top" feature. Did you ever end up implementing that? If so, I apologize for not being able to figure out how to pin an account. Thanks!

**Comment by OddHouseGames** [comment]
Just wanted to add my +1 to this one, Raivo is such an improvement over Free OTP with easy export and macOS reciever, only thing that could make it better is:

- 📂 Folders (Work, Personal, so they're not all mixed, and easier to update if change jobs etc..)
- 🔃 Ability to change sort ('most recent' instead of 'alphabetical'), for easy access without having to search)
- 📌 Ability to pin a few favs to the top for guaranteed quick access.

Feature requests aside, just want to say thanks and keep up the great work! 🙏

**Comment by Krychaz** [comment]
Love the app and its functionality, pretty much the only thing holding me back from fully switching to it is the lack of folders or some alternative. This is very useful when organizing your OTP's between work and personal.

**Comment by alonbl** [comment]
I just tried to switch, without having my common accounts grouped and visible it is very hard for me to switch. Please support at least manual reorder, then consider adding groups, tabs or other organization methods. Common scenario is to have most used account at visible location, and split personal and work in different tabs.

**Comment by erricodavide** [comment]
I would love to be able to manually order my OTPs (at least the pinned ones).
I'd like to migrate to Raivo entirely, but the lack of this simple, yet practical feature is holding me back.

**Comment by JCKE** [comment]
You can sort of make your own favourites by editing your entries to force them to the top, an underscore or period works: _ or .

Not a permanent solution by any means, but it stops me having to scroll everytime for the most-used services.

![Raivo-Renaming-Entries](https://user-images.githubusercontent.com/6626801/217986629-91e89773-63e6-4061-9abe-10081c0b1f50.jpeg)

**Comment by tijme** [comment]
> You can sort of make your own favourites by editing your entries to force them to the top, an underscore or period works: _ or .
> 
> 
> 
> Not a permanent solution by any means, but it stops me having to scroll everytime for the most-used services.
> 
> 
> 
> ![Raivo-Renaming-Entries](https://user-images.githubusercontent.com/6626801/217986629-91e89773-63e6-4061-9abe-10081c0b1f50.jpeg)
> 
> 

You can do this by pinning them to top, by swiping an OTP to the left.

**Comment by JCKE** [comment]
> You can do this by pinning them to top, by swiping an OTP to the left.

Didn't realize that, thanks! Manual sorting is still not there, renaming entries does help with that

**Comment by LawHoo** [comment]
First, Raivo OTP is an awesome app. Thank you! Second, I wanted to add a +1 for Folders. I have roughly 200 OTP accounts, and even though search is great, I'd really like to be able to group them into folders (e.g., Work, Client A, Client B, Personal, Family, Rarely Used, etc.). I'd also like an "All Accounts" view that shows all OTP accounts across all folders. In other words, if you add a folder feature, I would still want to be able to view all accounts in one large scrolling list and not be limited to ONLY viewing them in their respective folders.
